components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4970,4 +5013,31 @@ private void validateEnvironment(String organization, String envId) throws APIMa
         apiProvider.getEnvironment(organization, envId);
     }
 
+    /**
+     * This method checks whether updating API endpoints relates to service API endpoint URL belonging to the same
+     * organization or not
+     *
+     * @param properties   EnvironmentPropertiesDTO object
+     * @param apiId        API Id
+     * @param organization Organization name
+     * @return true if updating API endpoint is invalid, false otherwise
+     */
+    private static boolean isUpdatingApiEndpointConfigInvalid(EnvironmentPropertiesDTO properties, String apiId,
+                                                              String organization) {
+        boolean isEndpointInvalid = false;
+        if (properties.getProductionEndpoint() != null) {
+            isEndpointInvalid = RestApiPublisherUtils.isEndpointBelongingToAuthorizedOrg(
```

Comment:
Why do we have a conflicting variable name and a method name?
```suggestion
            isEndpointInvalid = RestApiPublisherUtils.isEndpointBelongingToAuthorizedOrg(
```